### PR TITLE
[BD-6] Python 3.8 compatibility. Upgrade requirements.

### PR DESCRIPTION
**Pull request by ericfab179**
Remove django constraints in base.in
Add constraints.
Update makefile upgrade command to create django.txt requirement file.
Add python 3.5 and 3.8 with Django 2.2 to the tests.
Remove Django 1.11 from tests.
Add venv to gitignore.

**Reviewers:**
- [x] @morenol
- [ ] @awais786

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ericfab179! I've created [OSPR-4469](https://openedx.atlassian.net/browse/OSPR-4469) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: tox.ini [hunk 1]
```diff
@@ -19,7 +19,8 @@ whitelist_externals =
     mkdir
     rm
 deps =
-    django111: Django>=1.11,<2
+    django22: -rrequirements/django.txt
+    django30: Django>3.0
```
Comment (mraarif) [inline tox.ini:23]: change the dependency specification for Django 3 to `django30: Django>=3.0,<3.1` for consistency

File: tox.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{27,35}-django111
+envlist = py{35}-django{22},py{38}-django{22,30},quality
```
Comment (mzulqarnain1) [inline tox.ini:2]: you can write it as `envlist = py35-django22,py38-django{22,30},quality`, no need for curly brackets when you have only one entity.

File: google_drive/google_calendar.py [hunk 3]
```diff
@@ -33,7 +33,7 @@ def _(text):
 
 # Classes ###########################################################
 @XBlock.needs("i18n")  # pylint: disable=too-many-ancestors
-class GoogleCalendarBlock(XBlock, PublishEventMixin):
+class GoogleCalendarBlock(XBlock, PublishEventMixin):  # pylint: disable=too-many-ancestors
```
Comment (mzulqarnain1) [inline google_drive/google_calendar.py:36]: you can disable this warning altogether by adding it in `pylintrc` file's disable section.

**Comment by ericfab179** [comment]
@awais786 We think this PR is ready to be merged.

**Comment by ericfab179** [comment]
@awais786 I think this PR is ready to be merged.

**Comment by OmarIthawi** [comment]
@ericfab179 Would please rebase this PR to kick off Travis builds again? I'm checking if I could review it.

@natabene I see that this PR ticket ([OSPR-4469](https://openedx.atlassian.net/browse/OSPR-4469)) have been moved to BLENDED-192.

What's the process here if I want to review it as an OSPR ticket?

**Comment by natabene** [comment]
@OmarIthawi You can review it here, within the Blended project, like you normally do with OSPRs.

**Comment by OmarIthawi** [comment]
@ericfab179 I'm going to review this pull request. I've done a quick pass and no major blockers which is great.

Please review rebase the pull request to kick off another TravisCI build.

**Comment by ericfab179** [comment]
@OmarIthawi I did the rebase and push but the tests seem to be still pending, could you please help me running them ? thanks

**Comment by OmarIthawi** [comment]
Thanks @ericfab179. I see that tests passes. I'll review it soon.

**Review by OmarIthawi — COMMENTED** [review]
Thanks @ericfab179. I have reviewed the changes and they look good.

 - 👍🏼 Review the code changes.
 - 👍🏼 Review the Travis/tox changes.
 - 👍🏼 Reviewed the requirement.
 - 👍🏼 Installed the XBlock on a Juniper devstack.
 - 👍🏼 Ran tox locally with the two python versions `tox -e py35-django22 -- google_drive/tests/unit` and `tox -e py38-django22 -- google_drive/tests/unit`

**Review by OmarIthawi — APPROVED** [review]
Thanks @ericfab179. I have reviewed the changes and they look good.

 - 👍🏼 Review the code changes.
 - 👍🏼 Review the Travis/tox changes.
 - 👍🏼 Reviewed the requirement.
 - 👍🏼 Installed the XBlock on a Juniper devstack.
 - 👍🏼 Ran tox locally with the two python versions `tox -e py35-django22 -- google_drive/tests/unit` and `tox -e py38-django22 -- google_drive/tests/unit`

**Comment by OmarIthawi** [comment]
@natabene I think this is good to get merged.

**Comment by OmarIthawi** [comment]
Sorry for the late notice @ericfab179 could you please change the commit message:

`Add too-many-ancestors to pylintrc.` ->  `[BD-6] Python 3.8 compatibility. Upgrade requirements.`

**Comment by OmarIthawi** [comment]
@ericfab179 could you please amend the commit title so it's more descriptive of the change?

**Comment by OmarIthawi** [comment]
@ericfab179 just a reminder to amend the commit title so it's more descriptive of the change.

**Comment by OmarIthawi** [comment]
Amazing! Thanks @ericfab179.

@natabene Could you please schedule that for merge? It's ready imho.

**Comment by natabene** [comment]
@OmarIthawi Thanks for reviewing. I don't manage BD tickets, I think @marcotuts can help us with moving this forward.

**Comment by OmarIthawi** [comment]
Thanks @natabene. @marcotuts, I've reviewed this PR with a couple of other PRs. It would be great if we can get it merged because otherwise `master` is broken and Travis won't run.

**Comment by awais786** [comment]
@jmbowman FYI

**Comment by jmbowman** [comment]
Thanks for sticking with this and getting it over the line!

**Pull request merged**